cmake/FindSHUM.cmake

```
@@ -34,10 +34,17 @@ To guide the search process, set the SHUM_ROOT environment variable or CMake var
 
 #]=======================================================================]
 
-find_path(SHUM_INCLUDE_DIR
-  NAMES f_shum_fieldsfile_class_version_mod.mod
-  PATH_SUFFIXES ${SHUM_PATH}/include
-)
+if(CMAKE_Fortran_COMPILER_ID STREQUAL "Cray")
+  find_path(SHUM_INCLUDE_DIR
+    NAMES F_SHUM_FIELDSFILE_CLASS_VERSION_MOD.mod
+    PATH_SUFFIXES ${SHUM_PATH}/include
+  )
+else()
+  find_path(SHUM_INCLUDE_DIR
+    NAMES f_shum_fieldsfile_class_version_mod.mod
+    PATH_SUFFIXES ${SHUM_PATH}/include
+  )
+endif()
```

Looks good. Can also consider this alternate form to handle any other similar variants in a single call. The NAMES argument, according to CMake documentation, _specify one or more possible names for the file in a directory_.

```suggestion
find_path(
  SHUM_INCLUDE_DIR
  NAMES F_SHUM_FIELDSFILE_CLASS_VERSION_MOD.mod f_shum_fieldsfile_class_version_mod.mod
  PATH_SUFFIXES ${SHUM_PATH}/include
)
```